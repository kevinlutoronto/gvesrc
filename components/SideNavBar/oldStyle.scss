@import '../../styles/base';

.wrapper, .itemWrapper {
  width: 100px;
}

.wrapper {
  display: flex;
  flex: 1;
  background: $cPurple100;
  @extend %type--tiny;
  color: rgba(255,255,255,0.8);
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: flex-start;
  padding-top: 8px;
  height: 100%;
  box-sizing: border-box;
}

.itemWrapper {
  align-self: flex-start;
  box-sizing: border-box;
  display: flex;
  padding: $space10;
  cursor: pointer;
  align-items: center;
  flex-direction: column;
  justify-content: center;
  text-align: center;
  font-size: 13px;
  position: relative;
  border-bottom: 1px solid rgba(0,0,0,0.3);
  transition: all 0.3s;
  border-left: 5px solid $cPurple100;
  border-right: 5px solid $cPurple100;

  &:hover, &:active, &:focus {
    background: $cPurple200;
    border-left-color: $cPurple200;
    border-right-color: $cPurple200;
  }

  .icon {
    margin-bottom: 5px;
  }
}

.darkBg {
  background: fade-out(#000, 0.9);
}

.selected {
  background: fade-out($cPurple200, 0.9);
  border-left-color: $cPurple200;
  border-right-color: #303a4c;
}