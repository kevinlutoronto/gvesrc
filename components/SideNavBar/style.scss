@import '../../styles/base';

.wrapper,
.groupWrapper {
  width: 180px;
}

.wrapper {
  font-family: $fontStack;
  background: $cGray600;
  height: 100%;
  box-sizing: border-box;

  a {
    color: $cBlack20;
    text-decoration: none;
  }
}

.top,
.input {
  font-size: 14px;
  color: $cGray20;
  font-weight: bold;
}

.top {
  box-sizing: border-box;
  background-color: $cPurple50;
  height: 50px;
  width: 100%;
  display: flex;
  justify-content: flex-start;
  align-items: center;
  cursor: pointer;
  &:hover,
  &:focus {
    color: $cGray40;
  }
  &:active {
    color: $cGray20;
  }

  &.fullWidth {
    width: 100vw;
  }
}

.sidebarContent {
  height: 100%;
  box-sizing: border-box;
  border-right: 1px solid $cBorder;
  padding: 20px 0 50px 0;
  overflow-y: scroll;

  &:focus {
    outline: none;
  }
}

.groupWrapper {
  box-sizing: border-box;
  padding: 0 10px 0 0px;
  margin-bottom: 12px;
  font-size: 14px;
  position: relative;
  transition: all 0.1s;
  box-sizing: border-box;
  color: $cBlack30;
  cursor: pointer;
}

.groupSubItems {
  margin-top: 2px;
}

.itemWrapper {
  box-sizing: border-box;
  padding: 6px 0px 6px 40px;
  font-size: 13px;
  position: relative;
  transition: all 0.1s;
  color: $cBlack20;
  box-sizing: border-box;
  display: block;
  &.selected,
  &.selected a {
    color: $cPurple30;
    &:hover {
      color: $cPurple30;
    }
  }
  &:hover,
  &:focus,
  a:hover {
    color: $cBlack40;
  }
  &:active,
  a:active {
    color: $cBlack50;
  }
}

.iconAndLabelContainer {
  position: relative;
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  align-items: center;
  overflow: hidden;

  .iconWrapper {
    width: 40px;
    min-width: 40px;
    display: flex;
    justify-content: center;
    align-items: center;
    position: relative;
    top: 1px;
  }

  .topText {
    color: $cGray20;
    text-overflow: ellipsis;
    white-space: nowrap;
    overflow: hidden;
    padding-right: 20px;
  }

  .groupLabelWrapper {
    color: $cPurple50;
    font-weight: bold;
    text-overflow: ellipsis;
    white-space: nowrap;
    overflow: hidden;
    padding-right: 20px;

    > * {
      color: $cPurple50;
    }
  }
}

.tag {
  position: absolute;
  right: 0;
  top: 5px;
  background-color: $cPurple30;
  color: $cGray20;
  font-size: 11px;
  padding: 3px 5px;
  border-radius: 2px;
  pointer-events: none;
}

.darkBg {
  background: fade-out(#000, 0.9);
}

.input {
  font-family: $fontStack;
  background-color: transparent;
  border: none;
  width: 100%;
  padding: 0px;

  &:focus {
    outline: none;
    color: $cGray20;
  }

  &::-webkit-input-placeholder,
  &::placeholder,
  &[placeholder] {
    text-overflow: ellipsis;
    white-space: nowrap;
  }
}

.fullWidth .input {
  width: 80vw;
}
