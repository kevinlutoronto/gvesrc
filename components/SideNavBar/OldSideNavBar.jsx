// @flow
import React, { Component } from 'react';
import style from './oldStyle.scss';

type Props = {|
  children: React$Node,
|};

class SideNavBar extends Component<Props> {
  render() {
    return <div className={style.wrapper}>{this.props.children}</div>;
  }
}

SideNavBar.displayName = 'Plasma@SideNavBar';

export default SideNavBar;
