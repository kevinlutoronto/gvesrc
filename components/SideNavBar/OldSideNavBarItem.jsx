// @flow
import React, { Component } from 'react';
import cn from 'classnames';
import style from './oldStyle.scss';

type Props = {|
  icon: string,
  iconSize: number,
  iconStyle: Object,
  label: string,
  onClick: () => mixed,
  selected: boolean,
  children: React$Node,
  darkBg: boolean,
|};

class SideNavBarItem extends Component<Props> {
  static defaultProps = {
    label: 'Label',
    iconSize: 24,
    selected: false,
  };

  renderIconAndLabel(): React$Node {
    const { icon, iconSize, iconStyle, label } = this.props;

    const renderIcon = icon && iconSize && (
      <img
        className={style.icon}
        style={{
          width: iconSize,
          height: iconSize,
          ...iconStyle,
        }}
        src={icon}
        alt={label}
      />
    );

    return (
      <div>
        {renderIcon}
        <div className={style.label}>{label}</div>
      </div>
    );
  }

  render() {
    const { children, darkBg, onClick, selected } = this.props;

    const wrapperClasses = cn(style.itemWrapper, {
      [style.darkBg]: darkBg,
      [style.selected]: selected,
    });

    return (
      <div className={wrapperClasses} onClick={onClick}>
        {children || this.renderIconAndLabel()}
      </div>
    );
  }
}

SideNavBarItem.displayName = 'Plasma@SideNavBarItem';

export default SideNavBarItem;
