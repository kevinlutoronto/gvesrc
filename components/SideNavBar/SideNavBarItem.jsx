// @flow
import React, { Component } from 'react';
import cn from 'classnames';
import style from './style.scss';

type Props = {|
  id: string,
  label: React$Node,
  onClick: string => mixed,
  selected: boolean,
  darkBg?: boolean,
  linkComponent: ?React$Node,
  tagText: ?string,
|};

class SideNavBarItem extends Component<Props> {
  static defaultProps = {
    label: 'Label',
    selected: false,
    tagText: null,
  };

  _onClick = (): void => {
    this.props.onClick(this.props.id);
  };

  renderIconAndLabel(): React$Node {
    const { label } = this.props;

    return (
      <div className={style.iconAndLabelContainer}>
        <div className={style.label}>{label}</div>
      </div>
    );
  }

  render() {
    const { darkBg, selected, linkComponent, tagText } = this.props;

    const wrapperClasses = cn(style.itemWrapper, {
      [style.darkBg]: darkBg,
      [style.selected]: selected,
    });

    let comp;

    if (linkComponent) {
      comp = (
        <div className={wrapperClasses}>
          {linkComponent}
          {tagText && <div className={style.tag}>{tagText}</div>}
        </div>
      );
    } else {
      comp = (
        <div className={wrapperClasses} onClick={this._onClick}>
          {this.renderIconAndLabel()}
          {tagText && <div className={style.tag}>{tagText}</div>}
        </div>
      );
    }

    return comp;
  }
}

SideNavBarItem.displayName = 'Plasma@SideNavBarItem';

export default SideNavBarItem;
