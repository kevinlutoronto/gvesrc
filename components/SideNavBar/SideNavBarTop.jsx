// @flow
import React from 'react';
import cn from 'classnames';
import Icon from '../Icon/Icon';
import style from './style.scss';

type Props = {|
  label: React$Node,
  icon: string,
  onClick: (SyntheticMouseEvent<>) => void,
  editing?: boolean,
  placeholder?: string,
  value?: string,
  onChange?: (SyntheticInputEvent<HTMLInputElement>) => void,
  onBlur?: (SyntheticFocusEvent<>) => void,
  fullWidth?: boolean,
  className?: string,
  forwardedRef?: { current: null | HTMLInputElement },
|};

class SideBarNavTop extends React.Component<Props> {
  input: { current: null | HTMLInputElement };

  // $FlowFixMe (https://github.com/facebook/flow/issues/6103)
  input = this.props.forwardedRef || React.createRef();

  componentDidUpdate(prevProps: Props) {
    if (this.props.editing && !prevProps.editing) {
      this.input.current && this.input.current.focus();
    }
  }

  render() {
    const {
      label,
      onClick,
      icon,
      editing,
      placeholder,
      value,
      onChange,
      onBlur,
      fullWidth,
      className,
    } = this.props;
    return (
      <div className={cn(className, style.top, { [style.fullWidth]: fullWidth })} onClick={onClick}>
        <div className={style.iconAndLabelContainer}>
          <div className={style.iconWrapper}>
            <Icon color="#fff" icon={icon} />
          </div>
          <div className={style.topText}>
            {editing ? (
              <input
                className={style.input}
                value={value}
                // $FlowFixMe (https://github.com/facebook/flow/issues/6103)
                ref={this.input}
                type="text"
                onChange={onChange}
                onBlur={onBlur}
                placeholder={placeholder}
              />
            ) : (
              label
            )}
          </div>
        </div>
      </div>
    );
  }
}

// $FlowFixMe
export default React.forwardRef((props, ref) => <SideBarNavTop {...props} forwardedRef={ref} />);
