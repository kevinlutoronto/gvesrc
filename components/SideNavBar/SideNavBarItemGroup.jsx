// @flow
import React, { Component } from 'react';
import { find } from 'lodash';
import SideNavBarItem from './SideNavBarItem';
import { type Item } from './SideNavBar';
import style from './style.scss';

type Props = {|
  id: string,
  icon?: string,
  iconSize: number,
  iconStyle?: Object,
  label: React$Node,
  onClick?: ({ id: string }) => mixed,
  items?: Array<Item>,
  selectedId?: string,
|};

class SideNavBarItemGroup extends Component<Props> {
  static defaultProps = {
    label: 'Label',
    iconSize: 22,
  };

  handleGroupClick = () => {
    if (this.props.onClick) {
      if (this.props.items && this.props.items[0]) {
        this.props.onClick({ id: this.props.items[0].id });
      } else {
        this.props.onClick({ id: this.props.id });
      }
    }
  };

  handleItemClick = (itemId: string): void => {
    if (this.props.onClick) {
      this.props.onClick({ id: itemId });
    }
  };

  renderIconAndLabel() {
    const { icon, iconSize, iconStyle, label, items } = this.props;
    if (!find(items, item => !item.hidden)) {
      return null;
    }

    const iconToRender = icon && iconSize && (
      <img
        className={style.icon}
        style={{
          width: iconSize,
          height: iconSize,
          ...iconStyle,
        }}
        src={icon}
        alt={label}
      />
    );

    return (
      <div onClick={this.handleGroupClick} className={style.iconAndLabelContainer}>
        <div className={style.iconWrapper}>{iconToRender}</div>
        <div className={style.groupLabelWrapper}>{label}</div>
      </div>
    );
  }

  renderSubItems(): ?React$Node {
    return (
      this.props.items &&
      this.props.items.map((groupedItem: Item) => {
        return (
          !groupedItem.hidden && (
            <SideNavBarItem
              key={groupedItem.id}
              id={groupedItem.id}
              label={groupedItem.label}
              tagText={groupedItem.tagText}
              onClick={this.handleItemClick}
              selected={groupedItem.id === this.props.selectedId}
              linkComponent={groupedItem.linkComponent}
            />
          )
        );
      })
    );
  }

  render() {
    return (
      <div className={style.groupWrapper}>
        {this.renderIconAndLabel()}
        <div className={style.groupSubItems}>{this.renderSubItems()}</div>
      </div>
    );
  }
}

SideNavBarItemGroup.displayName = 'Plasma@SideNavBarItemGroup';

export default SideNavBarItemGroup;
