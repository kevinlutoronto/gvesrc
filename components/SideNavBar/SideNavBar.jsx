// @flow
import React, { Component, type Node } from 'react';
import { getDataAttrs } from '../../dataUtils';
import type { Data } from '../../types';
import SideNavBarItemGroup from './SideNavBarItemGroup';
import SideNavBarTop from './SideNavBarTop';
import style from './style.scss';

export type Item = {|
  id: string,
  icon?: string,
  label: React$Node,
  items?: Array<Item>,
  linkComponent?: React$Node,
  tagText?: string,
  hidden?: boolean,
|};

type Props = {|
  items: Array<Item>,
  onChange?: ({ id: number }) => void,
  selectedId?: string,
  onClickTop?: () => void,
  topText?: string,
  topIcon?: string,
  data?: Data,
  editingTopText?: boolean,
  topTextValue?: string,
  onTopTextChange?: ({ value: string }) => void,
  onBlurTop?: () => void,
|};

class SideNavBar extends Component<Props> {
  handleClick = (result: Object): void => {
    this.props.onChange && this.props.onChange({ id: result.id });
  };

  renderItems(): Array<Node> {
    return this.props.items.map(item => (
      <SideNavBarItemGroup
        key={item.id}
        id={item.id}
        icon={item.icon}
        label={item.label}
        onClick={this.handleClick}
        items={item.items}
        selectedId={this.props.selectedId}
      />
    ));
  }

  render() {
    const {
      data,
      editingTopText,
      onBlurTop,
      onClickTop,
      onTopTextChange,
      topIcon,
      topText,
      topTextValue,
    } = this.props;

    return (
      <div {...getDataAttrs(data)} className={style.wrapper}>
        {(topText || topIcon) && (
          <SideNavBarTop
            label={topText}
            icon={topIcon}
            onClick={onClickTop}
            onBlur={onBlurTop}
            editing={editingTopText}
            placeholder={topText}
            value={topTextValue}
            onChange={onTopTextChange}
          />
        )}
        <div className={style.sidebarContent}>{this.renderItems()}</div>
      </div>
    );
  }
}

SideNavBar.displayName = 'Plasma@SideNavBar';

export default SideNavBar;
