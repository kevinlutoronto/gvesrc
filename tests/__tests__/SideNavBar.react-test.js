/* eslint-disable */

import React from 'react';
import SideNavBar from '../../components/SideNavBar/SideNavBar';
import SideNavBarTop from '../../components/SideNavBar/SideNavBarTop';
import SideNavBarItemGroup from '../../components/SideNavBar/SideNavBarItemGroup';
import SideNavBarItem from '../../components/SideNavBar/SideNavBarItem';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';

let items;

describe('SideNavBar Component', () => {
  beforeEach(() => {
    items = [
      {
        id: 'one',
        label: 'One',
        items: [
          {
            id: 'one-one',
            label: 'One One',
          },
        ],
      },
      {
        id: 'two',
        label: 'Two',
        items: [
          {
            id: 'two-one',
            label: 'Two One',
          },
          {
            id: 'two-hidden',
            label: 'Two Hidden',
            hidden: true,
          },
        ],
      },
      {
        id: 'three',
        label: 'Three',
      },
    ];
  });

  test('render', () => {
    const component = renderer.create(<SideNavBar items={items} />);
    expect(component).toMatchSnapshot();
  });

  test('onClickTop', () => {
    const onClick = jest.fn();
    const component = mount(<SideNavBar topText="YOLO" onClickTop={onClick} items={items} />);
    component.find(SideNavBarTop).simulate('click');
    expect(onClick).toHaveBeenCalledTimes(1);
  });

  test('hidden items', () => {
    const component = mount(<SideNavBar items={items} />);
    expect(
      component
        .find(SideNavBarItemGroup)
        .at(1)
        .find(SideNavBarItem)
    ).toHaveLength(1);
  });
});
