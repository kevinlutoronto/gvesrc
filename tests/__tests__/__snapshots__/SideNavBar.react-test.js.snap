// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideNavBar Component render 1`] = `
<div
  className="wrapper"
>
  <div
    className="sidebarContent"
  >
    <div
      className="groupWrapper"
    >
      <div
        className="iconAndLabelContainer"
        onClick={[Function]}
      >
        <div
          className="iconWrapper"
        />
        <div
          className="groupLabelWrapper"
        >
          One
        </div>
      </div>
      <div
        className="groupSubItems"
      >
        <div
          className="itemWrapper"
          onClick={[Function]}
        >
          <div
            className="iconAndLabelContainer"
          >
            <div
              className="label"
            >
              One One
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="groupWrapper"
    >
      <div
        className="iconAndLabelContainer"
        onClick={[Function]}
      >
        <div
          className="iconWrapper"
        />
        <div
          className="groupLabelWrapper"
        >
          Two
        </div>
      </div>
      <div
        className="groupSubItems"
      >
        <div
          className="itemWrapper"
          onClick={[Function]}
        >
          <div
            className="iconAndLabelContainer"
          >
            <div
              className="label"
            >
              Two One
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="groupWrapper"
    >
      <div
        className="groupSubItems"
      />
    </div>
  </div>
</div>
`;
